pkg/controller/rolloutblock/rolloutblock_controller.go

```diff
@@ -141,9 +145,17 @@ func (c *Controller) onUpdateRelease(oldObj interface{}, newObj interface{}) {
 	// rolloutblock object should be updated if a release has added it to override annotation,
 	// or removed it from override annotation
 	if len(oldOverrideRBs.Diff(newOverrideRBs)) > 0 {
-		c.enqueueReleaseBlock(oldObj)
+		if oldRel.GetResourceVersion() == newRel.GetResourceVersion() {
+			c.enqueueRolloutBlockKeysLater(oldOverrideRBs)
```

Comment:
Same principle of `*After` would apply in here (unless `*Later` is equivalent to `*After` with some configuration deep in the controller). If not, adding a `time.Duration` argument here also makes sense, and rename it to `...After`.

After reviewed the `enqueueRolloutBlockKeysLater()` implementation, it'd still make sense to pass a default duration to this method, allowing the caller to have control on the default computation of the duration.